src/Utils/Strict.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * This file is part of the Nette Framework (https://nette.org)
+ * Copyright (c) 2004 David Grudl (https://davidgrudl.com)
+ */
+
+namespace Nette;
+
+use Nette\Utils\ObjectMixin;
+
+
+/**
+ * Strict class (access to undeclared members throws exceptions) with 'did you mean' hints for better experience.
+ */
+trait Strict
+{
+
+	/**
+	 * Call to undefined method.
+	 * @throws MemberAccessException
+	 */
+	public function __call($name, $args)
+	{
+		$class = method_exists($this, $name) ? 'parent' : get_class($this);
+		$items = (new \ReflectionClass($this))->getMethods(\ReflectionMethod::IS_PUBLIC);
+		$hint = ($t = ObjectMixin::getSuggestion($items, $name)) ? ", did you mean $t()?" : '.';
+		throw new MemberAccessException("Call to undefined method $class::$name()$hint");
+	}
+
+
+	/**
+	 * Call to undefined static method.
+	 * @throws MemberAccessException
+	 */
+	public static function __callStatic($name, $args)
+	{
+		$rc = new \ReflectionClass(get_called_class());
+		$items = array_intersect($rc->getMethods(\ReflectionMethod::IS_PUBLIC), $rc->getMethods(\ReflectionMethod::IS_STATIC));
+		$hint = ($t = ObjectMixin::getSuggestion($items, $name)) ? ", did you mean $t()?" : '.';
+		throw new MemberAccessException("Call to undefined static method {$rc->getName()}::$name()$hint");
+	}
+
+
+	/**
+	 * Access to undeclared property.
+	 * @throws MemberAccessException
+	 */
+	public function &__get($name)
+	{
+		$rc = new \ReflectionClass($this);
+		$items = array_diff($rc->getProperties(\ReflectionProperty::IS_PUBLIC), $rc->getProperties(\ReflectionProperty::IS_STATIC));
+		$hint = ($t = ObjectMixin::getSuggestion($items, $name)) ? ", did you mean $$t?" : '.';
+		throw new MemberAccessException("Attempt to read undeclared property {$rc->getName()}::$$name$hint");
+	}
+
+
+	/**
+	 * Access to undeclared property.
+	 * @throws MemberAccessException
+	 */
+	public function __set($name, $value)
+	{
+		$rc = new \ReflectionClass($this);
+		$items = array_diff($rc->getProperties(\ReflectionProperty::IS_PUBLIC), $rc->getProperties(\ReflectionProperty::IS_STATIC));
+		$hint = ($t = ObjectMixin::getSuggestion($items, $name)) ? ", did you mean $$t?" : '.';
+		throw new MemberAccessException("Attempt to write to undeclared property {$rc->getName()}::$$name$hint");
+	}
+
+
+	/**
+	 * @return bool
+	 */
+	public function __isset($name)
```

Comment:
I was thinking about this earlier when I was [making this](https://github.com/Kdyby/StrictObjects/blob/b334bea700510844f6de31d2c2f6030f9ea3c26d/src/Scream.php#L89-L101) and I was unable to come up with benefit of returning `FALSE` here.
In Latte, it might make sense, since there are magic properties on the template object, but it does not make sense generaly imho.

If you're beeing trully strict, this potentially hides the fact, that you're doing an `isset()` check on property that is not defined.
Also, you probably never wanna do isset checks on classes that are strict, since you simply expect the property be in the definition.